### ISSUE TITLE
✅ 🐞 Element Caching causes WooCommerce shortcodes inside Text Editor widgets to be cached with user data

### ISSUE THREAD

**Issue by dtugend** [elementor/elementor#29613]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [ ] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

Element Cache BETA enabled by default in Elementor 3.26.0 (2024-12-16) causes payment issues at best and information leakage and people paying for wrong orders at worst

This feature is unstable and should not be enabled by default at all and an update should be issued to fix this asap.

*Setting this feature to inactive fixes the problem.*

Please not that when I generated the report I already disabled the feature.

### Steps to reproduce

1. Have a site e.g. with latest version of Elementor, Elementor Pro, Hello-Theme, WooCommerce and PeleCard gateway (somewhat popular in Israel).
2. Make a order using the PeleCard gateway.
3. Observe on the checkout page the order doesn't go through at best, or at worst you see payment data for different order.

### Expected behavior

This feature is unstable and should not be enabled by default at all and an update should be issued to fix this asap.

Ideally this should be fixed, but it might be difficult to fix it in all cases, so would be better to not have that feature.

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt


== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MySQL Community Server - GPL v8.0.40
	PHP Version: 7.4.33
	PHP Memory Limit: 256M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 32M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.7.1
	Site URL: https://www.veten.co.il
	Home URL: https://www.veten.co.il
	WP Multisite: No
	Max Upload Size: 16 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: he_IL
	Timezone: Asia/Jerusalem
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 3.2.0
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en-US
	User Agent: Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0

== Active Plugins ==
	Activity Log
		Version: 2.11.2
		Author: Activity Log Team

	Advanced Custom Fields
		Version: 6.3.11
		Author: WP Engine

	Advanced Order Export For WooCommerce
		Version: 3.5.6
		Author: AlgolPlus

	Checkout Field Editor for WooCommerce
		Version: 2.0.6
		Author: ThemeHigh

	Classic Editor
		Version: 1.6.7
		Author: WordPress Contributors

	CURCY - WooCommerce Multi Currency Premium
		Version: 2.3.5
		Author: VillaTheme

	Duplicator Pro
		Version: 4.5.20
		Author: Duplicator

	Elementor
		Version: 3.26.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.26.0
		Author: Elementor.com

	FluentSMTP
		Version: 2.2.83
		Author: FluentSMTP & WPManageNinja Team

	Head & Footer Code
		Version: 1.3.7
		Author: Aleksandar Urošević

	Header Footer Code Manager
		Version: 1.1.39
		Author: DraftPress

	Loco Translate
		Version: 2.6.14
		Author: Tim Whitlock

	MainWP Child
		Version: 5.3.1
		Author: MainWP

	One Click Accessibility
		Version: 2.1.0
		Author: One Click Accessibility

	Pelecard Gateway
		Version: 1.4.32
		Author: Pelecard

	ravpage
		Version: 2.31
		Author: Mati Skiba @ Rav Messer

	Really Simple Security
		Version: 9.1.4
		Author: Really Simple Security

	Redirection
		Version: 5.5.1
		Author: John Godley

	Shamor
		Version: 1.8.1
		Author: wpshamor.com

	Site Kit by Google
		Version: 1.142.0
		Author: Google

	Two Factor
		Version: 0.10.0
		Author: WordPress.org Contributors

	VGS Force Two-Factor
		Version: 1.0.11
		Author: Vagas VGS

	WC Disable Zoom / Lightbox features
		Version: 1.1
		Author: Rynaldo Stoltz

	woo-commbox-integration
		Version: 1.0
		Author: Eyal Ron

	WooCommerce
		Version: 9.4.3
		Author: Automattic

	WooCommerce Direct Checkout
		Version: 3.4.2
		Author: QuadLayers

	WooCommerce Legacy REST API
		Version: 1.0.4
		Author: WooCommerce

	WP All Export - WooCommerce Export Add-On Pro
		Version: 1.0.9
		Author: Soflyy

	WP All Export Pro
		Version: 1.9.1
		Author: Soflyy

	WPC Variations Radio Buttons for WooCommerce
		Version: 3.5.9
		Author: WPClever

	WP Super Cache
		Version: 1.12.4
		Author: Automattic

	Yoast Duplicate Post
		Version: 4.5
		Author: Enrico Battocchi & Team Yoast

	Yoast SEO
		Version: 24.0
		Author: Team Yoast

	YouTube WordPress Plugin by Embed Plus
		Version: 14.2.1.2
		Author: Embed Plus YouTube Plugin Team


== Must-Use Plugins ==
	Elementor Safe Mode
		Version: 1.0.0
		Author: Elementor.com

	ManageWP - Worker Loader
		Version: 1.0.0
		Author: GoDaddy


== Elements Usage ==
	
	page : 1
		divider : 1
		image : 1
		text-editor : 1
	section : 2
		button : 2
		divider : 1
		heading : 9
		html : 1
		image : 2
		social-icons : 1
		text-editor : 1
	wp-page : 30
		button : 6
		divider : 10
		heading : 41
		html : 1
		image : 20
		image-box : 7
		image-carousel : 2
		spacer : 1
		tabs : 1
		testimonial : 7
		text-editor : 30
		video : 9


== Settings ==
	
	cpt_support: post, page
	disable_color_schemes: yes
	disable_typography_schemes: yes
	allow_tracking: yes
	editor_break_lines: 1


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	woocommerce: Active


== Elementor Experiments ==
	Inline Font Icons: Inactive by default
	Additional Custom Breakpoints: Active by default
	Container: Inactive by default
	Upgrade Swiper Library: Active by default
	Nested Elements Performance: Active by default
	Optimized Markup: Inactive by default
	Plugin Onboarding: Active by default
	CSS Smooth Scroll: Active by default
	Default to New Theme Builder: Active by default
	Header & Footer: Inactive by default
	Elementor Home Screen: Active by default
	Landing Pages: Active by default
	Nested Elements: Active by default
	Editor Top Bar: Active by default
	Pages Panel: Inactive by default
	Element Caching: Inactive
	Link In Bio: Active by default
	Floating Buttons: Active by default
	Elementor Editor Events: Inactive by default
	Atomic Widgets: Inactive by default
	Global Classes: Inactive by default
	Launchpad Checklist: Inactive by default
	Menu: Inactive by default


== Log ==
	
PHP: showing 10 of 10PHP: 2024-11-05 09:49:47 [notice X 8][../wp-content/plugins/elementor-pro/modules/custom-css/module.php::76] trim(): Passing null to parameter #1 ($string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-18 11:49:47 [error X 8][../wp-content/plugins/elementor/modules/page-templates/module.php::270] Allowed memory size of 134217728 bytes exhausted (tried to allocate 102400 bytes) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-18 11:59:19 [error X 18][../wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php::173] Allowed memory size of 134217728 bytes exhausted (tried to allocate 81920 bytes) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-18 12:00:17 [error X 227][../wp-content/plugins/elementor/includes/base/controls-stack.php::2172] Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-18 12:35:20 [error X 2][../wp-content/plugins/elementor/includes/base/widget-base.php::612] Allowed memory size of 134217728 bytes exhausted (tried to allocate 196608 bytes) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-21 03:24:59 [error X 6][../wp-content/plugins/elementor/includes/base/element-base.php::442] Allowed memory size of 134217728 bytes exhausted (tried to allocate 16384 bytes) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-21 03:25:35 [error X 1][../wp-content/plugins/elementor-pro/modules/dynamic-tags/tags/author-url.php::12] LiteSpeed\Root::cls(): Cannot use output buffering in output buffering display handlers [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-21 05:00:25 [error X 10][../wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php::173] Allowed memory size of 134217728 bytes exhausted (tried to allocate 49152 bytes) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-21 10:58:29 [error X 1][../wp-content/plugins/elementor/core/settings/page/model.php::19] Allowed memory size of 134217728 bytes exhausted (tried to allocate 65536 bytes) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-12-02 23:52:18 [warning X 84][../wp-content/plugins/elementor/includes/conditions.php::90] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Log: showing 20 of 332024-12-02 17:56:45 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.7',
  'to' => '3.25.4',
)]
2024-12-11 01:27:35 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'אלמנטור',
  'from' => '3.25.10',
  'to' => '3.25.11',
)]
2024-12-13 07:44:08 [info] elementor-pro::elementor_pro_updater Started 
2024-12-13 07:44:08 [info] אלמנטור פרו/Upgrades - _on_each_version Start  
2024-12-13 07:44:08 [info] אלמנטור פרו/Upgrades - _on_each_version Finished 
2024-12-13 07:44:08 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'אלמנטור פרו',
  'from' => '3.25.4',
  'to' => '3.25.5',
)]
2024-12-13 07:44:08 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'אלמנטור פרו',
  'from' => '3.25.4',
  'to' => '3.25.5',
)]
2024-12-13 08:10:06 [info] elementor::elementor_updater Started 
2024-12-13 08:10:06 [info] Elementor/Upgrades - _on_each_version Start  
2024-12-13 08:10:06 [info] Elementor/Upgrades - _on_each_version Finished 
2024-12-13 08:10:06 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.10',
  'to' => '3.25.11',
)]
2024-12-13 08:10:07 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.10',
  'to' => '3.25.11',
)]
2024-12-17 11:22:14 [info] elementor::elementor_updater Started 
2024-12-17 11:22:14 [info] Elementor/Upgrades - _on_each_version Start  
2024-12-17 11:22:14 [info] Elementor/Upgrades - _on_each_version Finished 
2024-12-17 11:22:14 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.11',
  'to' => '3.26.0',
)]
2024-12-17 12:23:54 [info] elementor-pro::elementor_pro_updater Started 
2024-12-17 12:23:54 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-12-17 12:23:54 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-12-17 12:23:54 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.25.5',
  'to' => '3.26.0',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatibility not specified

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by milosmaksimus** [comment]
Similar issue with latest update,

My account page becomes impossible to login to. 
If you try to access pages in MY ACCOUNT by link, everyone gets logged in like admin. 
Rolling back to version 3.25.8 fixed the issue.

**Comment by tobias992** [comment]
Welcome to the world of Elementor. You no longer need to worry about your settings - elementor will do it for you. It's just cheeky and totally unacceptable to change your settings like that. Especially when I have disabled an experiment, I expect that setting to be respected.

I have no idea where this is going...

**Comment by SupremeSimon** [comment]
Yeah, this update also caused our property search page on 10s of sites to stop being able to paginate as the page was cached...

**Comment by tomandersen** [comment]
TOTAL FAIL

Example: All Woocommerce /my-account/ pages that are done in Elementor are now broken. Customers cannot login! 

"Element Caching active by default on existing websites: This optimization improves loading speeds by caching static content automatically." see also https://github.com/elementor/elementor/issues/28372

**Comment by tomandersen** [comment]
You need to patch and turn it off NOW

**Comment by tomandersen** [comment]
It is very easy to reproduce. Make woocommerce site. Elementor the /my-account/ page, then its broken. Acessing the /my-account/ page will show the cached logged in screen of some user, or a login screen that appears not to work, - whatever is in the cache. 

The cache is holding onto confidential information from other users and displaying it to the world.

**Comment by nicholaszein** [comment]
Hello @dtugend and @milosmaksimus!

Thank you for reaching out and sharing your feedback! 🙏 

We are investigating this issue and actively working on a solution.

From visiting your websites we can see that in both cases you were using the respective WooCommerce **`shortcodes`** for your WooCommerce pages (e.g.: **`[woocommerce_checkout]`** for the WooCommerce **Checkout page**, and the **`[woocommerce_my_account]`** for the **My Account page**) inside **Elementor Text Editor widgets** instead of using the [WooCommerce Checkout](https://elementor.com/help/woocommerce-checkout-widget/) and WooCommerce [My Account](https://elementor.com/help/woocommerce-my-account-widget/) widgets included with Elementor Pro.

We are trying to understand the reason behind this choice, and if you even knew these widgets were available and recommended to be used with Elementor when these pages are being edited with Elementor.

Can you explain a little more about your setup and the choice to go with the shortcodes inside the Text Editor?

Thank you for your time and I hope to hear from you soon.

Kind regards

**Comment by dtugend** [comment]
Hi. Thank you for your reply.

We did not build the particular site in question, but from past experience with Elementor shop sites we built and other page builders it's usually plugin compatibility issues with some WooCommerce addons that used to arise when using e.g. Elementor widgets, not sure if the situation has improved but that's the reason we did it too back then.

WooCommerce emits PHP constants and headers that prevent caching for most caching plugins, adhering to these might dampen the problems at least for WordPress plugins aware of caching:
- https://github.com/woocommerce/woocommerce/blob/04c63f02fee90a7ff2a9a20874ce20205379eeb7/plugins/woocommerce/includes/class-wc-cache-helper.php#L247
(Of course won't help much for other plugins with dynamic shortcodes that are not aware.)

**Comment by lankathilina** [comment]
@nicholaszein - We’ve encountered similar issues on multiple WooCommerce-based sites that use Elementor as the page editor, particularly when WooCommerce shortcodes are used for My Account and Checkout pages.

My Account Page:
When entering login details and clicking the login button, the page loops back to the same login screen.
On one occasion, I was logged in as a user without actually logging in (though I haven’t been able to replicate this since).
Switching from WooCommerce shortcodes to Elementor widgets seemed to resolve the issue for some sites, but the problem persists on some.

Checkout Page:
When a user attempts to log in on the checkout page, nothing happens, and the page loops back.
This issue occurs on almost all affected sites. Disabling Elementor’s element cache has resolved the problem in these cases.

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core v3.26.2! 🥳 

**We reverted back the activation of Element Caching on existing websites.**

To learn how to make the most of Elementor with Element Caching Active and improve your website's performance without breaking dynamic content inserted via shortcodes or dynamic tags, please read the document below and learn how to set Element Caching as **Inactive** individually per widget while still leveraging it on other widgets without completely deactivating the feature.

https://elementor.com/help/element-caching-help/

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
